jsonnet/patch-rules.libsonnet

```diff
@@ -1,36 +1,21 @@
-local patchRule(rule, patchedRules) =
-  if std.length(patchedRules) == 0 then
-    [rule]
-  else if (('alert' in rule && 'alert' in patchedRules[0]) && rule.alert == patchedRules[0].alert) ||
-          (('record' in rule && 'record' in patchedRules[0]) && rule.record == patchedRules[0].record) then
-    [std.mergePatch(rule, patchedRules[0])]
-  else
-    [] + patchRule(rule, patchedRules[1:]);
-
-local patchRuleGroup(group, patchedGroups) =
-  if std.length(patchedGroups) == 0 then
-    [group.rules]
-  else if (group.name == patchedGroups[0].name) then
-    [patchRule(rule, patchedGroups[0].rules) for rule in group.rules]
-  else
-    [] + patchRuleGroup(group, patchedGroups[1:]);
+local patchOrExcludeRuleGroup = (import 'utils.libsonnet').patchOrExcludeRuleGroup;
 
 {
   patchRules(o): {
```

Comment:
You can put this in one file with `excludeRules`. In [`main.libsonnet` we are referencing a specific field during import](https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/main.jsonnet#L3), so both can live in the same file and reduce complexity. This means we can have `patchRules`, `excludeRules`, and everything from `utils.libsonnet` in the same place :)